packages/providers/src.ts/base-provider.ts

```diff
@@ -751,44 +750,70 @@ export class BaseProvider extends Provider {
      *
      * @param txId - id of the transaction to search for
      */
-    async getTransaction(txId: string | Promise<string>): Promise<TransactionResponse> {
+    async getTransaction(transactionId: string | Promise<string>): Promise<TransactionResponse> {
+        //currently considers only previewnet!
         await this.getNetwork();
-        txId = await txId;
-        const ep = '/api/v1/transactions/'+txId;
-        let { data } = await axios.get(this._mirrorNodeUrl + ep);
-        const filtered = data.transactions
-            .filter((e: { result: string; }) => e.result === "SUCCESS");
-        return filtered.length > 0 ? filtered[0] : null;
+        if (!this._mirrorNodeUrl) logger.throwError("missing provider", Logger.errors.UNSUPPORTED_OPERATION);
+        transactionId = await transactionId;
+        //subsequent requests depend on finalized transaction
+        const epTransactions = '/api/v1/transactions/' + transactionId;
+        try {
+            let { data } = await axios.get(this._mirrorNodeUrl + epTransactions);
+            let response;
+            if (data) {
+                const filtered = data.transactions.filter((e: { result: string; }) => e.result != 'DUPLICATE_TRANSACTION');
+                const res = filtered.length > 0 ? filtered[0] : null;
+                if (res) {
+                    const epContracts = '/api/v1/contracts/results/' + transactionId;
+                    response = Promise.all([
+                        axios.get(this._mirrorNodeUrl + epContracts)
+                    ])
+                        .then(async ([contracts]) => {
+                            const mergedData = { ...contracts.data, transaction: { transaction_id: res.transaction_id, result: res.result }};
+                            return this.formatter.txRecordToTxResponse(mergedData);
+                        })
+                        .catch(error => {
+                            console.log(error);
+                            return null;
+                        });
+                }
+            } 
+            return response;
+        } catch (error) {
+            if (error.response.status != 404) {
+                logger.throwError("bad result from backend", Logger.errors.SERVER_ERROR, {
+                    method: "TransactionResponseQuery",
+                    error
+                });
+            }
+            return null;
+        }
     }
 
-    async getTransactionReceipt(transactionHash: string | Promise<string>): Promise<TransactionReceipt> {
+    //TODO this will not be supported? 
```

Comment:
It will be supported, but in a future PR i guess.
nit: remove the comment or annotate it for future implementation